File: src/main/factory/trafficSource/listTrafficSourceByIdFactory.ts
import { PrismaTrafficSourceRepository } from "../../../infra/repositories/trafficSource";
import { ListTrafficSourceByIdUseCase } from "../../../app/useCases/trafficSource/listTrafficSourceByIdUseCase";
import { ListTrafficSourceByIdController } from "../../../infra/controllers/trafficSource/listTrafficSourceByIdController";
import { UserGateway } from "../../../infra/gateways/user";

const prismaTrafficSourceRepository = new PrismaTrafficSourceRepository();
const userGateway = new UserGateway();

const listTrafficSourceByIdUseCase = new ListTrafficSourceByIdUseCase(
  prismaTrafficSourceRepository,
  userGateway
);

const listTrafficSourceByIdController = new ListTrafficSourceByIdController(
  listTrafficSourceByIdUseCase
);

const listTrafficSourceById = {
  handle: listTrafficSourceByIdController.handle.bind(
    listTrafficSourceByIdController
  ),
};

export { listTrafficSourceById };

File: internal/main/factory/trafficSource/listTrafficSourceByIdFactory.go
package trafficSourceFactories

import (
	trafficSourceUseCases "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/app/useCases/trafficSource"
	"github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/domain/repositories"
	trafficSourceControllers "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/infra/controllers/trafficSource"
	contracts "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/main/contracts"
)

func NewListTrafficSourceByIdFactory(trafficSourceRepository repositories.TrafficSourceRepository) contracts.Controller {
	listTrafficSourceByIdUseCase := trafficSourceUseCases.NewListTrafficSourceByIdUseCase(trafficSourceRepository)
	return trafficSourceControllers.NewListTrafficSourceByIdController(listTrafficSourceByIdUseCase)
}
